### PR TITLE
Oil 186/soft blocking tags

### PR DESCRIPTION
**Pull request by ltparis2018**
New:
- Soft blocking of tags marked as "managed tag" using 'data-managed="as-oil"' attribute
- Corresponding unit and end-to-end tests
- Corresponding demo page

Bugfix:
- wait for successful (asynchronous soi and poi) cookie setting before executing following cookie-dependent actions

### CODE REVIEW

File: src/scripts/core/core_tag_management.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import { MANAGED_TAG_IDENTIFIER, MANAGED_TAG_IDENTIFIER_ATTRIBUTE, MANAGED_TAG_PURPOSES_ATTRIBUTE } from './core_constants';
+import { getSoiCookie } from './core_cookies';
+import { arrayContainsArray } from './core_utils';
+import { getPurposeIds } from './core_vendor_information';
+
+export function manageDomElementActivation() {
+  let managedElements = findManagedElements();
+  let cookie = getSoiCookie();
+
+  for (let i = 0; i < managedElements.length; i++) {
```
Comment (subsend) [inline src/scripts/core/core_tag_management.js]: managedElements.length perfomance technisch besser ausserhalb for schleife.

File: src/scripts/core/core_tag_management.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    manageScriptElement(element, cookie);
+  } else {
+    manageNonScriptElement(element, cookie);
+  }
+}
+
+function manageScriptElement(element, cookie) {
+  let newElement = document.createElement('script');
+
+  for (let i = 0; i < element.attributes.length; i++) {
```
Comment (subsend) [inline src/scripts/core/core_tag_management.js]: same wie in Zeile 10'

File: src/scripts/core/core_tag_management.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  newElement.async = element.async;
+  newElement.charset = element.charset;
+
+  let parent = element.parentElement;
+  parent.insertBefore(newElement, element);
+  parent.removeChild(element);
+}
+
+function manageNonScriptElement(element, cookie) {
+  let managedAttributes = ['href', 'src', 'title', 'display'];
```
Comment (subsend) [inline src/scripts/core/core_tag_management.js]: display ist auch html tag attribute?

File: test/specs/core/core_tag_management.spec.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+import * as CoreCookies from '../../../src/scripts/core/core_cookies';
+import * as CoreVendorInformation from '../../../src/scripts/core/core_vendor_information';
+import { loadFixture, readFixture } from '../../test-utils/utils_fixtures';
+import { manageDomElementActivation } from '../../../src/scripts/core/core_tag_management';
+
+const {ConsentString} = require('consent-string');
+
+describe("Tag Management", () => {
+
+  // The following consent strings represent
```
Comment (CanardSauvage) [inline test/specs/core/core_tag_management.spec.js]: hihi I like

File: src/scripts/userview/userview_modal.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import '../../styles/modal.scss';
 import '../../styles/cpc.scss';
-import {sendEventToHostSite} from '../core/core_utils.js';
-import {removeSubscriberCookies} from '../core/core_cookies.js';
+import { sendEventToHostSite } from '../core/core_utils.js';
+import { removeSubscriberCookies } from '../core/core_cookies.js';
```
Comment (subsend) [inline src/scripts/userview/userview_modal.js]: import einheitlich entweder mit js oder ohne :)

File: src/scripts/core/core_utils.js [hunk 6]
```diff
@@ -129,6 +129,26 @@ export function arrayContains(array, obj) {
   return false;
 }
 
+export function arrayContainsArray(array, checkedArray) {
+  if (!array || !checkedArray) {
+    return false;
+  }
+  let elementCounterMap = {};
+
+  for (let i = 0; i < array.length; i++) {
```
Comment (subsend) [inline src/scripts/core/core_utils.js]: length

File: src/scripts/core/core_utils.js [hunk 7]
```diff
@@ -129,6 +129,26 @@ export function arrayContains(array, obj) {
...
+    if (!elementCounterMap[array[i]]) {
+      elementCounterMap[array[i]] = 0;
+    }
+    elementCounterMap[array[i]]++;
+  }
+  for (let i = 0; i < checkedArray.length; i++) {
```
Comment (subsend) [inline src/scripts/core/core_utils.js]: length

**Comment by CanardSauvage** [comment]
Very good work.

**Pull request merged**